index.js

```diff
@@ -54,11 +55,23 @@ function decorateLayer(layer, keysToKeep, newProps) {
         keepLookup[keys.indexOf(keysToKeep[i])] = true;
     }
 
+    if (requiredKeys && requiredKeys.length) {
+        for (i = 0; i < requiredKeys.length; i++) {
+            var index = keys.indexOf(requiredKeys[i]);
+            if (index !== -1) requiredIndices.push(index);
+        }
+    }
```

Comment:
benchtest for decorate went from 32ms to 20ms after updating hasAllKeys to use a hashmap (ed80768) 👍